### PR TITLE
Creating property on non-config slot should throw type error.

### PR DESCRIPTION
**Pull request by akroshg**
Based on the spec, when we try to set property on the non-configurable property slot should throw type error. Fixed that multiple places.

### CODE REVIEW

**Comment by akroshg** [comment]
@suwc - please have a look.

**Comment by akroshg** [comment]
fixes #757.

**Comment by suwc** [comment]
:shipit:

**Comment by suwc** [comment]
LGTM.

File: lib/Parser/rterrors.h [hunk 1]
```diff
@@ -374,3 +374,4 @@ RT_ERROR_MSG(WASMERR_InvalidGlobalRef, 5673, "", "Global initialization does not
 RT_ERROR_MSG(WASMERR_NeedMemoryObject, 5674, "%s is not a WebAssembly.Memory", "WebAssembly.Memory object expected", kjstTypeError, 0)
 RT_ERROR_MSG(WASMERR_InvalidTypeConversion, 5675, "Invalid WebAssembly type conversion %s to %s", "Invalid WebAssembly type conversion", kjstTypeError, 0)
 RT_ERROR_MSG(WASMERR_DivideByZero, 5676,    "",  "Division by zero", kjstError, 0)
+RT_ERROR_MSG(JSERR_CantRedefineProp, 5677, "Cannot redefine property '%s'", "Cannot redefine property", kjstTypeError, 0)
```
Comment (agarwal-sandeep) [inline lib/Parser/rterrors.h]: >JSERR_CantRedefineProp [](start = 13, length = 22)

How is this different from error message JSERR_DefineProperty_NotConfigurable (Cannot redefine non-configurable property). Are there cases when we hit this error and the property is configurable?

File: lib/Runtime/Library/JavascriptArray.cpp [hunk 2]
```diff
@@ -3032,7 +3040,7 @@ namespace Js
                             }
                             else
                             {
-                                SetArrayLikeObjects(pDestObj, idxDest, subItem);
+                                ThrowErrorOnFailure(SetArrayLikeObjects(pDestObj, idxDest, subItem), scriptContext, idxSubItem);
```
Comment (agarwal-sandeep) [inline lib/Runtime/Library/JavascriptArray.cpp]: >idxSubItem [](start = 116, length = 10)

Should this be idxDest? if not can we put a comment explaining why its idxSubItem as we are setting the element at idxDest.

File: lib/Parser/rterrors.h (hunk 1 above)
Comment (akroshg) [inline lib/Parser/rterrors.h]: That was my first choice as well, but I am trying to match the wording with other browser. Also SetItem can fail for other reasons, so it is better to be not-specific to configurable only.

---
In reply to: [87445471](https://github.com/Microsoft/ChakraCore/pull/1969#discussion_r87445471) [](ancestors = 87445471)

File: lib/Runtime/Library/JavascriptArray.cpp (hunk 2 above)
Comment (akroshg) [inline lib/Runtime/Library/JavascriptArray.cpp]: It should be idxDest, nice catch. will fix that.

---
In reply to: [87446064](https://github.com/Microsoft/ChakraCore/pull/1969#discussion_r87446064) [](ancestors = 87446064)

**Pull request merged**